lib/discordrb/data/interaction.rb

```diff
@@ -398,6 +398,108 @@ def edit(name: nil, description: nil, default_permission: nil, &block)
     def delete
       @bot.delete_application_command(@id, server_id: @server_id)
     end
+
+    # Get the permission configuration for the this application command on a specific server.
+    # @param server_id [Integer, String, nil] The ID of the server to fetch command permissions for.
+    # @return [Array<Permission>] the permissions for this application command in the given server.
+    def permissions(server_id: nil)
+      raise ArgumentError, 'A server ID must be provided for global application commands' if @server_id.nil? && server_id.nil?
+
+      response = JSON.parse(API::Application.get_application_command_permissions(@bot.token, @bot.profile.id, @server_id || server_id&.resolve_id, @id))
+      response['permissions'].map { |permission| Permission.new(permission.merge({ '_command' => response.except('permissions') }), @bot) }
+    rescue Discordrb::Errors::UnknownError
+      # If there aren't any explicit overwrites configured for the command, the response is a 400.
+      []
+    end
+
+    # An application command permission for a channel, member, or a role.
+    class Permission
+      # Map of permission types.
+      TYPES = {
+        role: 1,
+        member: 2,
+        channel: 3
+      }.freeze
+
+      # @return [Integer] the type of this permission.
+      # @see TYPES
+      attr_reader :type
+
+      # @return [Integer] the ID of the thing this permission is for.
+      attr_reader :target_id
+
+      # @return [Integer] the ID of the server this permission is for.
+      attr_reader :server_id
+
+      # @!visibility private
+      def initialize(data, bot)
+        @bot = bot
+        @type = data['type']
+        @target_id = data['id'].to_i
+        @overwrite = data['permission']
+        @command_id = data['_command']['id'].to_i
+        @server_id = data['_command']['guild_id'].to_i
+        @application_id = data['_command']['application_id'].to_i
```

Comment:
It looks like this is ephemeral (based on line 409), I would prefer to pass this as a second argument instead of embeding it into the data. `def initialize(data, command, bot)`